src/Temporalio/Bridge/CancellationToken.cs

```diff
@@ -8,6 +8,8 @@ namespace Temporalio.Bridge
     /// </summary>
     internal class CancellationToken : SafeHandle
     {
+        private System.Threading.CancellationTokenRegistration? registration;
```

Comment:
Can this be readonly and set in the constructor? Also, do we only create this via `FromThreading`? If so, I wonder if the constructor can be made private and this not be nullable (or turn `FromThreading` into the public constructor).